docs/topics/native/apple-framework.md

```diff
@@ -1,363 +1,329 @@
+where `Demo` is the framework name set by the `-output` parameter of `kotlinc-native`.
 
 ```objc
-NS_ASSUME_NONNULL_BEGIN
-
 __attribute__((objc_subclassing_restricted))
-__attribute__((swift_name("LibKt")))
-@interface DemoLibKt : KotlinBase
-+ (void)forIntegersB:(int8_t)b s:(int16_t)s i:(int32_t)i l:(DemoLong * _Nullable)l __attribute__((swift_name("forIntegers(b:s:i:l:)")));
+__attribute__((swift_name("HelloKt")))
+@interface DemoHelloKt : DemoBase
++ (NSString * _Nullable)acceptFunF:(NSString * _Nullable (^)(NSString *))f __attribute__((swift_name("acceptFun(f:)")));
 + (void)forFloatsF:(float)f d:(DemoDouble * _Nullable)d __attribute__((swift_name("forFloats(f:d:)")));
++ (void)forIntegersB:(int8_t)b s:(uint16_t)s i:(int32_t)i l:(DemoULong * _Nullable)l __attribute__((swift_name("forIntegers(b:s:i:l:)")));
 + (NSString *)stringsStr:(NSString * _Nullable)str __attribute__((swift_name("strings(str:)")));
-+ (NSString * _Nullable)acceptFunF:(NSString * _Nullable (^)(NSString *))f __attribute__((swift_name("acceptFun(f:)")));
 + (NSString * _Nullable (^)(NSString *))supplyFun __attribute__((swift_name("supplyFun()")));
-@end;
+@end
 ```
 
-You see that Kotlin `String` and Objective-C `NSString*` are mapped transparently.
-Similarly, `Unit` type from Kotlin is mapped to `void`. We see primitive types
-are mapped directly. Non-nullable primitive types are mapped transparently.
-Nullable primitive types are mapped into `Kotlin<TYPE>*` types, as shown in the table [above](#kotlin-numbers-and-nsnumber). 
-Both higher order functions `acceptFunF` and `supplyFun` are included,
-and accept Objective-C blocks.
+Kotlin `String` and Objective-C `NSString*` are mapped transparently. Similarly, `Unit` type from Kotlin is mapped to `void`.
+The primitive types are mapped directly. Non-nullable primitive types are mapped transparently.
+Nullable primitive types are mapped into `Kotlin<TYPE>*` types, as shown in the [table](#kotlin-numbers-and-nsnumber). 
+Both higher-order functions `acceptFunF` and `supplyFun` are included and accept Objective-C blocks.
 
-More information about all other types mapping details can be found in the
-[Objective-C Interop](native-objc-interop.md)
-documentation article
+You can find more information about type mapping in [Interoperability with Swift/Objective-C](native-objc-interop.md#mappings).
 
 ## Garbage collection and reference counting
 
-Objective-C and Swift use reference counting. Kotlin/Native has its own garbage collection too.
-Kotlin/Native garbage collection is integrated with Objective-C/Swift reference
-counting. You do not need to use anything special to control the lifetime of Kotlin/Native instances
-from Swift or Objective-C.
+Objective-C and Swift use reference counting. Kotlin/Native has its own [garbage collector](native-memory-manager.md#garbage-collector)
+too. Kotlin/Native garbage collection is integrated with Objective-C/Swift reference counting.
```

Comment:
This is a nice place to give a link to https://kotlinlang.org/docs/native-arc-integration.html